src/flick/show/views.py

```diff
@@ -27,54 +36,72 @@ class ShowViewSet(mixins.ListModelMixin, mixins.RetrieveModelMixin, viewsets.Gen
     # if api_settings.UNPROTECTED, then any user can see this
     permission_classes = api_settings.STANDARD_PERMISSIONS
 
-    # don't need this, generics has this code, but this overrides
-    # gives option to add additional checks / customize
-    # def list(self, request):
-    #     # can access logged in user via request.user
-    #     self.serializer_class = ShowSerializer
-    #     return super(ItemList, self).list(request)
 
-    # def retrieve(self, request, pk):
-    #     queryset = self.get_object()
-    #     serializer = ShowDetailSerializer(queryset, many=False)
-    #     return success_response(serializer.data)
+class SearchShow(APIView):
+    shows = []
+    known_shows = []
 
+    # show_type can be "movie", "tv", "anime"
+    def get_shows_by_type_and_query(self, query, show_type, source):
+        show_ids = local_cache.get((query, show_type))
+        if not show_ids:
+            show_ids = API.search_show_ids_by_name(show_type, query)
+            local_cache.set((query, show_type), show_ids)
+        for show_id in show_ids:
+            known_show = Show.objects.filter(ext_api_id=show_id, ext_api_source=source)
+            if known_show.exists():
+                known_show = Show.objects.get(ext_api_id=show_id, ext_api_source=source)
+                self.known_shows.append(ShowSerializer(known_show).data)
+            else:
+                show = API.get_show_info_from_id(show_type, show_id)
+                if show:
+                    self.shows.append(show)
 
-class SearchShow(APIView):
+    def get_shows_by_query(self, query, is_movie, is_tv, is_anime):
+        if is_movie:
+            self.get_shows_by_type_and_query(query, "movie", "tmdb")
+        if is_tv:
+            self.get_shows_by_type_and_query(query, "tv", "tmdb")
+        if is_anime:
+            self.get_shows_by_type_and_query(query, "anime", "animelist")
 
-    permission_classes = api_settings.UNPROTECTED
+    def get_top_shows_by_type(self, show_type):
+        top_shows = local_cache.get(("top", show_type))
+        if not top_shows:
+            top_shows = API.get_top_show_info(show_type)
+            local_cache.set(("top", show_type), top_shows)
+        self.shows.extend(top_shows)
+
+    def get_top_shows(self, is_movie, is_tv, is_anime):
+        if is_movie:
+            self.get_top_shows_by_type("movie")
+        if is_tv:
+            self.get_top_shows_by_type("tv")
+        if is_anime:
+            self.get_top_shows_by_type("anime")
 
     def get(self, request, *args, **kwargs):
         query = request.query_params.get("query")
-        is_anime = request.query_params.get("is_anime")
-        is_movie = request.query_params.get("is_movie")
-        is_tv = request.query_params.get("is_tv")
-        is_top = request.query_params.get("is_top")
+        print(f"query: {query}")
```

Comment:
should we get rid of print statements

Reply:
good eye hehe but this was deliberate because I need to investigate why the queries are giving me True when i put the values in as false -- i think as long as you put the key in the value will be true (i'll explain this during our meeting!!)